frontend-react/src/components/ManagePublicKey/ManagePublicKey.tsx

```diff
@@ -1,18 +1,148 @@
-import React from "react";
-import { GridContainer } from "@trussworks/react-uswds";
+import React, { useState } from "react";
+import { useNavigate } from "react-router-dom";
+import { GridContainer, Icon, SiteAlert } from "@trussworks/react-uswds";
 
 import { AuthElement } from "../AuthElement";
 import { withCatchAndSuspense } from "../RSErrorBoundary";
+import { USLink } from "../USLink";
+import Spinner from "../Spinner";
+import { showError } from "../AlertNotifications";
 import { MemberType } from "../../hooks/UseOktaMemberships";
+import {
+    validateFileType,
+    ManagePublicKeyAction,
+    CONTENT_TYPE,
+    FORMAT,
+    validateFileSize,
+} from "../../hooks/network/ManagePublicKey/ManagePublicKeyHooks";
+
+import { ManagePublicKeyChooseSender } from "./ManagePublicKeyChooseSender";
+import { ManagePublicKeyUpload } from "./ManagePublicKeyUpload";
+
+const LightbulbIcon = Icon.Lightbulb;
+
+const ManagePublicKeySwitchDisplay = () => {
+    const navigate = useNavigate();
+    const [action, setAction] = useState(ManagePublicKeyAction.SELECT_SENDER);
+    const [fileContent, setFileContent] = useState("");
+    const [fileName, setFileName] = useState("");
+
+    // TODO: mocked for now - make the call you need when sending the file
+    const { sendFile } = {
+        sendFile: (data: {
+            contentType?: string;
+            fileContent?: string;
+            fileName?: string;
+        }) => {
+            data = { fileName: "filename" };
+            return data;
+        },
+    };
+
+    const onSenderSelect = (selectedSender: string) => {
+        if (selectedSender !== "") {
+            setAction(ManagePublicKeyAction.SELECT_PUBLIC_KEY);
+        }
+    };
+
+    const onBack = () => {
+        navigate("/resources");
```

Comment:
I don't think we'll always want to go back to /resources from this page, right?  Shouldn't this just be literally going back in navigation history?